src/obr/create_tree.py

```diff
@@ -199,6 +200,24 @@ def add_variations(
             }
             statepoint["parent"] = base_dict
 
+            # check for statepoint filters
+            if (
+                isinstance(value, dict)
+                and value.get("if", False)
+                and isinstance(value["if"], dict)
+            ):
+                statepoint_filter = value["if"].get("statepoint", {})
+                key = statepoint_filter["key"]
```

Comment:
This and the next line would throw an error if the previous line sets statepoint_filter to {}